### PR TITLE
ENH: Add CI.

### PR DESCRIPTION
**Pull request by jhlegarreta**
- Add `*.yml` configuration files for CI.
- Add the `setup.py` Python setup file.
- Add Python wrapping files.
- Add the `CTestConfig.cmake` file to make the build results be sent to
  the **Insight** project in **open.cdash.org**.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@phcerdan @thewtex The python wrapping will need a second pass/check. Thanks.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      - store_artifacts:
+          path: dist
+          destination: dist
+
+workflows:
+    version: 2
+    build-test-package:
+      jobs:
+        - build-and-test
+        - package
```
Comment (dzenanz) [inline .circleci/config.yml]: No newline at and of file. Is that an issue?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+build: off
+
+test: off
+
+artifacts:
+
+  # pushing entire folder as a zip archive
+  - path: dist\*
+
+deploy: off
```
Comment (dzenanz) [inline appveyor.yml]: no newline

**Comment by dzenanz** [comment]
Great progress. I guess it will be easier to tackle other issues once the CI is working.

File: appveyor.yml (hunk 2 above)
Comment (jhlegarreta) [reply appveyor.yml]: Thanks @dzenanz  ! Done in 6c736b7.

File: .circleci/config.yml (hunk 1 above)
Comment (jhlegarreta) [reply .circleci/config.yml]: Thanks @dzenanz  ! Done in 6c736b7.

**Comment by jhlegarreta** [comment]
Yes, CI is essential to improve our tools and allow us to focus on performance, quality of the code, etc. ! Getting closer to being able to measure all this at some point ! Thanks @dzenanz !

**Comment by thewtex** [comment]
Absolutely awesome @jhlegarreta .

**Pull request merged**